edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/MecanumDrive/src/Robot.cpp

```diff
@@ -1,54 +1,39 @@
+#include <IterativeRobot.h>
 #include <Joystick.h>
 #include <RobotDrive.h>
-#include <SampleRobot.h>
-#include <Timer.h>
 
 /**
  * This is a demo program showing how to use Mecanum control with the RobotDrive
  * class.
  */
-class Robot: public frc::SampleRobot {
+class Robot: public frc::IterativeRobot {
 public:
-	Robot() {
-		robotDrive.SetExpiration(0.1);
-
+	void RobotInit() {
 		// Invert the left side motors
-		robotDrive.SetInvertedMotor(RobotDrive::kFrontLeftMotor, true);
-
 		// You may need to change or remove this to match your robot
+		robotDrive.SetInvertedMotor(RobotDrive::kFrontLeftMotor, true);
 		robotDrive.SetInvertedMotor(RobotDrive::kRearLeftMotor, true);
 	}
 
-	/**
-	 * Runs the motors with Mecanum drive.
-	 */
-	void OperatorControl() override {
-		robotDrive.SetSafetyEnabled(false);
-		while (IsOperatorControl() && IsEnabled()) {
-			/* Use the joystick X axis for lateral movement, Y axis for forward
-			 * movement, and Z axis for rotation. This sample does not use
-			 * field-oriented drive, so the gyro input is set to zero.
-			 */
-			robotDrive.MecanumDrive_Cartesian(stick.GetX(), stick.GetY(),
-					stick.GetZ());
-
-			frc::Wait(0.005); // wait 5ms to avoid hogging CPU cycles
-		}
+	void TeleopPeriodic() override {
+		/* Use the joystick X axis for lateral movement, Y axis for forward
+		 * movement, and Z axis for rotation.
```

Comment:
It was confusing because we are using the method that does not take a gyro input.  There is an overloaded method that does take a gyro input.